### DISCUSSION TITLE
Problem with 3D Mortar Contact

### DISCUSSION THREAD

**Discussion by Flolaffel** [idaholab/moose#20734] [Q&A General] [answered]
Hello,

I'm experimenting with contact simulations and wanted to try out frictional Mortar for my 3D Model. But when I tried to run the calculation it exited with a `Segfault (Signal 11)`. So I decided to update MOOSE to see if that helps. Now the Segfault is gone but I get another error right at the beginning of the first time step:

```
/home/florian/miniconda3/envs/moose/libmesh/include/metaphysicl/dynamic_std_array_wrapper.h, line 125, compiled Apr  6 2022 at 18:07:01
application called MPI_Abort(MPI_COMM_WORLD, 1) - process 2
```

This problem is only present with 3D Mortar. I tested frictional and frictionless contact both setup with and without Contact Action but to no success. 2D Mortar seems to work just fine. 

What causes this error to appear?

**Comment by GiudGiud** [comment, marked as answer]
Hello
It s likely the AD container size that is too small.
Could you please configure with a larger derivative container size then recompile?
```
./configure --with-derivative-size=200
cd modules/contact
make
```

Guillaume

**Reply by Flolaffel** [reply]
That worked, thanks a lot.
